IGC/Compiler/Optimizer/IntDivConstantReduction.cpp

```diff
@@ -479,8 +483,28 @@ struct IntDivConstantReduction : public FunctionPass
         //
         const int bitSize = dividend->getType()->getIntegerBitWidth();
         //
+#if LLVM_VERSION_MAJOR >= 14
+        SignedDivisionByConstantInfo appxRecip = SignedDivisionByConstantInfo::get(divisor);
+#else
         APInt::ms appxRecip = divisor.magic();
+#endif
         //
+#if LLVM_VERSION_MAJOR >= 14
+        ConstantInt *appxRcp = IGC::getConstantSInt(
+            B, bitSize, appxRecip.Magic.getSExtValue());
+        Value *appxQ =
+            IGC::CreateMulh(F, B, true, dividend, appxRcp);
+        if (divisor.isStrictlyPositive() && appxRecip.Magic.isNegative()) {
+            appxQ = B.CreateAdd(appxQ, dividend, "q_appx");
+        }
+        if (divisor.isNegative() && appxRecip.Magic.isStrictlyPositive()) {
+            appxQ = B.CreateSub(appxQ, dividend, "q_appx");
+        }
+        if (appxRecip.ShiftAmount > 0) {
+            ConstantInt *shift = IGC::getConstantSInt(B, bitSize, appxRecip.ShiftAmount);
+            appxQ = B.CreateAShr(appxQ, shift, "q_appx");
+        }
+#else
```

Comment:
This seems like more complex example of potential wrapper usage. Having this amount of code under `#if LLVM` isn't desirable.

Maybe there's a way to generalize this under `llvmWrapper/Support/DivisionByConstantInfo.h`?